### ISSUE TITLE
The `config.yml` template comments need to be improved to avoid indentation failures

### ISSUE THREAD

**Issue by alberpilot** [wazuh/wazuh-packages#1860]
A community user[ had a problem](https://wazuh.slack.com/archives/C0A933R8E/p1664190144769509) using the `config.yml` template, as the example used [here](https://documentation.wazuh.com/current/user-manual/certificates.html) could mislead due to amount of characters to remove when using a distributed configuration. Example:

```
  server:
    - name: wazuh-1
      ip: <wazuh-manager-ip>
    # node_type: master
    # - name: wazuh-2
    #   ip: <wazuh-manager-ip>
    # node_type: worker
```

If a user removes the `#` character and one blank space, this is the result:
```
  server:
    - name: wazuh-1
      ip: <wazuh-manager-ip>
    node_type: master
    - name: wazuh-2
      ip: <wazuh-manager-ip>
    node_type: worker
```
`node_type` is now un-indented, which will produce problems generating certificates. 

## Tasks
- [ ] The `config.yml` template needs to be reviewed in order to find a more straightforward out-of-error format. 
- [ ] Determine if wazuh-ansible/docker/puppet etc needs to propagate the change. 
- [ ] wazuh-documentation PR changing the template. 
- [ ] Apply changes in production. 

# Stages (Updated)
Due to important cascade changes, it's better to split this issue into two jobs:
The tasks are still the same, but the changes should affect only to commented lines. Improving the user experience when removing those comments avoiding indentation errors. 

Changes described in comments like https://github.com/wazuh/wazuh-packages/issues/1860#issuecomment-1285030675 and https://github.com/wazuh/wazuh-packages/issues/1860#issuecomment-1290059536 will be completed in a new issue: https://github.com/wazuh/wazuh-packages/issues/1920

**Comment by davidjiglesias** [comment]
I would like to suggest a few other improvements to ease the user's understanding of the different modules:

From current:

```
nodes:
  # Wazuh indexer nodes
  indexer:
    - name: node-1
      ip: <indexer-node-ip>
  # Wazuh server nodes
  # Use node_type only with more than one Wazuh manager
  server:
    - name: wazuh-1
      ip: <wazuh-manager-ip>
  # Wazuh dashboard node
  dashboard:
    - name: dashboard
      ip: <dashboard-node-ip>
```

**Comment by s-ocando** [comment]
Hi @wazuh/cicd, 

Changing the default common names from `node-n` to `indexer-n` requires updating the [/etc/wazuh-indexer/opensearch.yml](https://github.com/wazuh/wazuh-packages/blob/4.3/stack/indexer/base/files/etc/wazuh-indexer/opensearch.yml) configuration file. 

The default names should be updated in the following fields: `node.name`, `cluster.initial_master_nodes`, `discovery.seed_hosts` and `plugins.security.nodes_dn`. 

```
network.host: "0.0.0.0"
node.name: "node-1"
cluster.initial_master_nodes:
- "node-1"
#- "node-2"
#- "node-3"
cluster.name: "wazuh-cluster"
#discovery.seed_hosts:
#  - "node-1-ip"
#  - "node-2-ip"
#  - "node-3-ip"
node.max_local_storage_nodes: "3"
path.data: /var/lib/wazuh-indexer
path.logs: /var/log/wazuh-indexer

plugins.security.ssl.http.pemcert_filepath: /etc/wazuh-indexer/certs/indexer.pem
plugins.security.ssl.http.pemkey_filepath: /etc/wazuh-indexer/certs/indexer-key.pem
plugins.security.ssl.http.pemtrustedcas_filepath: /etc/wazuh-indexer/certs/root-ca.pem
plugins.security.ssl.transport.pemcert_filepath: /etc/wazuh-indexer/certs/indexer.pem
plugins.security.ssl.transport.pemkey_filepath: /etc/wazuh-indexer/certs/indexer-key.pem
plugins.security.ssl.transport.pemtrustedcas_filepath: /etc/wazuh-indexer/certs/root-ca.pem
plugins.security.ssl.http.enabled: true
plugins.security.ssl.transport.enforce_hostname_verification: false
plugins.security.ssl.transport.resolve_hostname: false

plugins.security.authcz.admin_dn:
- "CN=admin,OU=Wazuh,O=Wazuh,L=California,C=US"
plugins.security.check_snapshot_restore_write_privileges: true
plugins.security.enable_snapshot_restore_privilege: true
plugins.security.nodes_dn:
- "CN=node-1,OU=Wazuh,O=Wazuh,L=California,C=US"
#- "CN=node-2,OU=Wazuh,O=Wazuh,L=California,C=US"
#- "CN=node-3,OU=Wazuh,O=Wazuh,L=California,C=US"
plugins.security.restapi.roles_enabled:
- "all_access"
- "security_rest_api_access"

plugins.security.system_indices.enabled: true
plugins.security.system_indices.indices: [".opendistro-alerting-config", ".opendistro-alerting-alert*", ".opendistro-anomaly-results*", ".opendistro-anomaly-detector*", ".opendistro-anomaly-checkpoints", ".opendistro-anomaly-detection-state", ".opendistro-reports-*", ".opendistro-notifications-*", ".opendistro-notebooks", ".opensearch-observability", ".opendistro-asynchronous-search-response*", ".replication-metadata-store"]

### Option to allow Filebeat-oss 7.10.2 to work ###
compatibility.override_main_response_version: true
```
Besides updating the corresponding templates, this change will require updating the examples throughout the installation guide, including instructions, Wazuh indexer's outputs, among others. 

This change also needs to be included in the [Wazuh indexer migration guide](https://documentation.wazuh.com/current/migration-guide/wazuh-indexer.html). Currently, the user is instructed to change the `OU` from `Wazuh` to `Docu` as in the old certificates. Now, we also need to instruct the user to change from `indexer-n` to `node-n` to match the old certificates common names.

**Comment by alberpilot** [comment]
Approved Wazuh packages PR: https://github.com/wazuh/wazuh-packages/pull/1901#pullrequestreview-1165989898
Blocked by https://github.com/wazuh/wazuh-documentation/pull/5727
